src/zeusops_bot/cogs/zeus_upload.py

```diff
@@ -50,6 +59,55 @@ async def zeus_upload(
                 f"Error was: {str(e)}"
             )
 
+    @commands.slash_command(name="zeus-upload")
+    async def zeus_upload_with_modlist(
+        self,
+        ctx: discord.ApplicationContext,
+        scenario_id: str,
+        modlist: DiscordAttachment,
+        filename: str,
+    ):
+        """Upload a mission as a Zeus, with modlist too"""
+        try:
+            with BytesIO() as inmemoryfile:
+                modlist.save(inmemoryfile)
+                modlist_json = json.loads(inmemoryfile.getvalue())
+            modlist_typeadapter.validate_python(modlist_json)
+            path = self.reforger_confgen.zeus_upload(
+                scenario_id, filename, modlist=modlist_json
+            )
+            await ctx.respond(f"Mission uploaded successfully under {path=}")
+        except json.JSONDecodeError as e:
+            await ctx.respond(
+                "Failed to understand the attached modlist as JSON. "
+                "Check the file was exported from the workshop, "
+                "try confirming with an online validator like "
+                "https://jsonlint.com/;"
+                f"Parse error was: {e}"
+            )
+        except ValidationError as e:
+            await ctx.respond(
+                "Failed to understand the modlist given: valid JSON, "
+                "but not a list of mod objects (name/ID/optional-version). "
+                "Check the file was exported from the workshop? "
+                f"Validation error was: {e}"
+            )
+        except ConfigFileNotFound:
+            await ctx.respond(
+                "Bot config error: the base config file could not be found"
+                f" Tell the Techmins! Path was: {self.reforger_confgen.base_config}"
+            )
+        except ConfigFileInvalidJson as e:
+            await ctx.respond(
+                "Bot config error: the base config file is invalid JSON "
+                "Tell the Techmins! Error was: " + str(e)
+            )
+        except ConfigPatchingError as e:
+            await ctx.respond(
+                "Failed to patch your requested change over base config.\n"
+                f"Error was: {str(e)}"
+            )
```

Comment:
I realize all these are identical to above, hmm, may want to have a single common bit for this "ending" once the modlist is sorted out?

Reply:
Combined both upload commands into a single command with an optional attachment, so the error handling isn't duplicated anymore